proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java

```diff
@@ -494,7 +494,11 @@ public void run()
                             }
                             if ( server == null )
                             {
-                                server = bungee.getServerInfo( listener.getDefaultServer() );
```

Comment:
Actually you never set the server in the event befoe you call the event and execute this code only if there is no server set by bungeecord itself.
You should provide the server the user is going to be sent, then call the event and then checking the event target always.